File: data-transform/geocode.js
var csv = require('csv');
var fs = require('fs');
var geocoder = require('geocoder');

var outStream = fs.createWriteStream('temp.csv');

var parser = csv.parse({delimiter: ',', columns: true}, function(crashErr, crashData){
	for (var i = 0; i < crashData.length; i++) {
		var intersectionString = crashData[i]["Primary Road"] + " and " + crashData[i]["Intersecting Road"] + " " + crashData[i]["Community"];
		geocoder.geocode(intersectionString, function ( geoErr, geoData ) {
			var outString = "";
			csv.stringify(crashData[i], function(err, output){
			  outString = output;
			});


			csv.stringify(crashData[i]);
			console.log(outString);//get rate limit errors hereeee :'( 
			if (geoData.results.length > 0){
				outString += "," + geoData.results[0].geometry.location.lat + "," + geoData.results[0].geometry.location.lng + "\n";
				outStream.write(outString);
				//console.log(geoData.results[0].formatted_address  + " : " + geoData.results[0].geometry.location.lat + " , " + geoData.results[0].geometry.location.lng);
			}
		});	
		//need to make this in a promise and slow it down to stop running into google's rate limit
	}
});

fs.createReadStream('../Bicyclist-Motorist_Crashes_2000-2013.csv').pipe(parser);

File: data-transform/geocode.rb
require 'rubygems'
require 'geocoder'
require 'csv'


#Geocoder.configure(:api_key => "", :use_https => true)

outfile = File.open('temp.csv', 'w')
outfile.write("Crash ID,County,Community,Primary Road,Intersecting Road,Date,Time,Type,Level,Crash Rpt.,long,lat\n")

CSV.foreach("../Bicyclist-Motorist_Crashes_2000-2013.csv", {:headers=>:first_row}) do |row|
	intersectionString = row["Primary Road"] + " and " + row["Intersecting Road"] + " " + row["Community"];

	coords = Geocoder.coordinates(intersectionString)

	outfile.write(row.to_csv(row_sep: nil) + "," + coords[0].to_s + "," +  coords[1].to_s + "\n");
	sleep(0.2) #slow it down to get around the rate limit
end
outfile.close